### DISCUSSION TITLE
blog/print-xaringan-chromote/

### DISCUSSION THREAD

**Discussion by giscus** [gadenbuie/garrickadenbuie-com#109] [Comments]
# blog/print-xaringan-chromote/

Create PDF versions of complicated xaringan slides using {chromote} and a little magic.

https://www.garrickadenbuie.com/blog/print-xaringan-chromote/

**Comment by anon35930** [comment]
Hi Garrick. Thank you so much for this ... it's exactly what I need. I have xaringan rendered local slides with path, for example, "file:///pathtofile.html".

When I attempt to execute the xaringan_to_pdf function, I get the following error: 

"Error in cpp_pdf_combine(input, output, password) : 
  open file:/pathtofile.pdf: No such file or directory"

Wondering if you would be able to help. Thanks in advance for your time!

**Reply by gadenbuie** [reply]
Have you seen [xaringanBuilder](https://jhelvy.github.io/xaringanBuilder/)? The code in this post now lives in that package where there have definitely been a few improvements. I'd check that out first and if you still have some problems, you can open an issue in that package repo and we'll be happy to help!

**Reply by anon35930** [reply]
I saw it after posting my question and it works perfectly. thank you again

**Comment by djorquerap** [comment]
Hi! thanks for the function, I made the PDF no problem... except one: the text inside some ggplots (that were previously rendered as SVG files) was outlined (see: https://community.adobe.com/t5/acrobat-discussions/letter-l-looks-bolded-in-pdf/td-p/8842120). They were not like this in the html file, so it was something in your function that made the change. Is there a way to avoid this?

**Comment by jorgesinval** [comment]
I tried to use it: `xaringan_to_pdf(input = "presentation.html", output_file = "presentation.pdf")` but...
`Error: Chromote: timed out waiting for event Page.loadEventFired`

**Reply by gadenbuie** [reply]
Hi! Can you try using the [renderthis](https://github.com/jhelvy/renderthis) package instead? The `xaringan_to_pdf()` function now lives on there as `renderthis::to_pdf()`.

**Reply by jorgesinval** [reply]
Thank you, but error:

```
✔ Rendering ../../Post_doc-Evidence-Based_Health/Cochrane Handbook/presentation.Rmd into ../…
Warning: A runtime exception has occured while executing JavaScriptpresentation.html into ..…
  Runtime exception message:
    ReferenceError: $ is not defined
    at http://127.0.0.1:4530/presentation_files/kePrint/kePrint.js:1:1
✖ Rendering ../../Post_doc-Evidence-Based_Health/Cochrane Handbook/presentation.html into ..…
Error in force(expr) : Failed to generate output in 30 seconds (timeout).
```

**Reply by gadenbuie** [reply]
Please open an issue with a reproducible example at [jhelvy/renderthis](https://github.com/jhelvy/renderthis/issues).
